Gems/Atom/Tools/ShaderManagementConsole/Code/Source/Window/ShaderManagementConsoleWindow.cpp

```diff
@@ -37,6 +38,62 @@ namespace ShaderManagementConsole
         Base::OnDocumentOpened(documentId);
         m_documentInspector->SetDocumentId(documentId);
     }
+
+    void ShaderManagementConsoleWindow::UpdateMenus(QMenuBar* menuBar)
+    {
+        Base::UpdateMenus(menuBar);
+        m_actionSaveAsChild->setVisible(false);
+    }
+
+    AZStd::string ShaderManagementConsoleWindow::GetSaveDocumentParams(const AZStd::string& initialPath) const
+    {
+        // Get shader file path
+        AZStd::string shaderFullPath;
```

Comment:
Instead of accepting an `const AZStd::string&`, this function could accept an AZ::IO::PathView, which is a Path API wrapper with `string_view` backend type. I also recommend that this function return an AZ::IO::Path as well.

Also the shaderFullPath can be made an `AZ::IO::Path` and a reference to it's underlying string type can be accessed via the [AZ::IO::Path::Native](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/IO/Path/Path.h#L542-L545) function.

```suggestion
        AZ::IO::Path shaderFullPath;
```

Reply:
Update of `GetSaveDocumentParams` is not part of this PR.
I will discuss with the author.